libr/debug/p/native/linux/linux_debug.c

```diff
@@ -509,3 +514,1397 @@ RList *linux_desc_list (int pid) {
+        phdr.p_memsz = 0x0;
+        phdr.p_align = 0x1;
+
+	ret = r_buf_append_bytes (dest, (const ut8 *)&phdr, sizeof (Elf64_Phdr));
+	if (ret != true) {
+		printf ("dump_elf_pheaders: r_buf_append_bytes error!\n");
+		free (note_data);
+		return false;
+    	}
+
+        offset_to_next = offset_to_note + note_section_size;
+
+        /* write program headers */
+
+        for (me_p = sec_note->maps; me_p != NULL; me_p = me_p->n) {
+                if (!(me_p->perms & R_MEM) && !(me_p->perms & W_MEM))
+                        continue;
+                phdr.p_type = PT_LOAD;
+                phdr.p_flags = me_p->perms;
+                phdr.p_vaddr = me_p->start_addr;
+                phdr.p_paddr = 0x0;
+                phdr.p_memsz = me_p->end_addr - me_p->start_addr;
+                phdr.p_filesz = me_p->dumpeable == 0 ? 0 : phdr.p_memsz;
+                phdr.p_offset = offset_to_next;;
+                phdr.p_align = 0x1;
+
+                offset_to_next += phdr.p_filesz == 0 ? 0 : phdr.p_filesz;
+
+		ret = r_buf_append_bytes (dest, (const ut8*)&phdr, sizeof (Elf64_Phdr));
+		if (!ret) {
+			printf ("dump_elf_pheaders: r_buf_append_bytes error!\n");
+			free (note_data);
+			return false;
+		}
+
+                memset (&phdr, '\0', sizeof(Elf64_Phdr));
+        }
+
+	eprintf ("pheaders writen\n");
+
+	ret = r_buf_append_bytes (dest, (const ut8*)note_data, note_section_size);
+	if (!ret) {
+		printf ("dump_elf_pheaders: r_buf_append_bytes error!\n");
+		free (note_data);
+		return false;
+	}
+
+        eprintf ("note writen\n");
+
+        return true;
+}
+
+
+static void show_maps(linux_map_entry_t *head) {
+        linux_map_entry_t *p;
+
+        eprintf ("SHOW MAPS ===================\n");
+        for (p = head; p ; p = p->n) {
+                if (p->name)
+                        eprintf ("p->name: %s\n", p->name);
+
+                eprintf ("p->start_addr - %lx, p->end_addr - %lx\n", p->start_addr, p->end_addr);
+        }
+        eprintf ("SHOW MAPS ===================\n");
+}
+
+static bool dump_elf_map_content(RBuffer *dest, linux_map_entry_t *head, pid_t pid) {
+        linux_map_entry_t *p;
+        struct iovec local;
+        struct iovec remote;
+        char *map_content;
+        size_t size;
+        size_t rbytes;
+
+        for (p = head; p != NULL ; p = p->n) {
+
+                eprintf ("\n\nTrying to dump: %p - %p\n", p->name, p->start_addr);
+
+		if (p->dumpeable) {
+
+			size = p->end_addr - p->start_addr;
+			map_content = malloc (size);
+			if (map_content == NULL) {
+				printf ("dump_elf_map_content: map_content == NULL\n");
+ 	       			return false;
+ 	       		}
+
+			eprintf ("p->name: %s - %p to %p - size: %d\n", p->name, p->start_addr, map_content, size);
+
+			local.iov_base = (void *)map_content;
+			local.iov_len = size;
+
+			remote.iov_base = (void *)p->start_addr;
+			remote.iov_len = size;
+
+			rbytes = process_vm_readv (pid, &local, 1, &remote, 1, 0);
```

Comment:
`warning: implicit declaration of function ‘process_vm_readv’`
You probably need to include `<sys/uio.h>`